src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3072,17 +3070,16 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             }
             else
             {
-                TypeSymbolWithAnnotations leftType = leftResult.Type;
-                TypeSymbolWithAnnotations rightType = ApplyConversion(right, right, conversion, leftType.TypeSymbol, rightResult.Type, checkConversion: true, fromExplicitCast: false, out bool canConvertNestedNullability);
+                TypeSymbolWithAnnotations rightType = ApplyConversion(right, right, conversion, leftType.TypeSymbol, rightResult, checkConversion: true, fromExplicitCast: false, out bool canConvertNestedNullability);
                 // Need to report all warnings that apply since the warnings can be suppressed individually.
                 ReportNullReferenceAssignmentIfNecessary(right, leftType, rightType, UseLegacyWarnings(left));
                 if (!canConvertNestedNullability)
                 {
-                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, right.Syntax, GetTypeAsDiagnosticArgument(rightResult.Type?.TypeSymbol), leftType.TypeSymbol);
+                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, right.Syntax, GetTypeAsDiagnosticArgument(rightResult?.TypeSymbol), leftType.TypeSymbol);
                 }
-                TrackNullableStateForAssignment(right, leftType, leftResult.Slot, rightType, rightResult.Slot);
+                TrackNullableStateForAssignment(right, leftType, MakeSlot(left), rightType, MakeSlot(right));
```

Comment:
>MakeSlot(left) [](start = 65, length = 14)

Are there observable side-effects from moving the slot creation for `left` after visiting the right?

Reply:
Or is it just slots could be reordered and we don't care about that?

---
In reply to: [198932012](https://github.com/dotnet/roslyn/pull/28155#discussion_r198932012) [](ancestors = 198932012)

Reply:
The order that slots are created should not matter.

---
In reply to: [198932781](https://github.com/dotnet/roslyn/pull/28155#discussion_r198932781) [](ancestors = 198932781,198932012)